dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/AbstractImageRegistry.java

```diff
@@ -519,15 +526,73 @@ private void updateDockerHubImageInformation(List<Tag> newTags, @NotNull Tool to
+
+        try {
+            URL projectURL = new URL(projectPath);
+            projectResponse = Optional.of(IOUtils.toString(projectURL, "UTF-8"));
+
+            if (projectResponse.isPresent()) {
+                final String projectJSON = projectResponse.get();
+                Gson gson = new Gson();
+                Type gitLabContainerRegistryListType = new TypeToken<ArrayList<GitLabContainerRegistry>>() { }.getType();
+                List<GitLabContainerRegistry> registries = gson.fromJson(projectJSON, gitLabContainerRegistryListType);
+
+                final String tagsListPath = projectPath + '/' + registries.get(0).getId() + '/' + "tags";
+                URL tagsListURL = new URL(tagsListPath);
+                Optional<String> tagsListResponse = Optional.of(IOUtils.toString(tagsListURL, "UTF-8"));
+
+                if (tagsListResponse.isPresent()) {
+                    String tagsListJSON = tagsListResponse.get();
+                    Type gitLabTagType = new TypeToken<ArrayList<GitLabTag>>() { }.getType();
+                    List<GitLabTag> gitLabTags = gson.fromJson(tagsListJSON, gitLabTagType);
+
+                    try {
+                        for (GitLabTag gitLabTag : gitLabTags) {
+                            final String detailedTagInfoUrlString = tagsListPath + '/' + gitLabTag.getName();
+                            URL detailedTagInfoURL = new URL(detailedTagInfoUrlString);
+                            Optional<String> detailedTagInfoResponse = Optional.of(IOUtils.toString(detailedTagInfoURL, "UTF-8"));
+
+                            if (detailedTagInfoResponse.isPresent()) {
+                                final String detailedTagInfoJSON = detailedTagInfoResponse.get();
+                                gitLabTag = gson.fromJson(detailedTagInfoJSON, GitLabTag.class);
+                                final Tag tag = new Tag();
+                                tag.setName(gitLabTag.getName());
+                                final String manifestDigest = gitLabTag.getDigest();
+                                List<Checksum> checksums = new ArrayList<>();
+
+                                checksums.add(new Checksum(manifestDigest.split(":")[0], manifestDigest.split(":")[1]));
+                                Set<Image> images = Collections.singleton(new Image(checksums, tool.getNamespace() + '/' + tool.getName(), tag.getName(), null));
+                                tag.getImages().addAll(images);
+                                tags.add(tag);
+                            }
+                        }
+                    } catch (IndexOutOfBoundsException | NullPointerException ex) {
+                        LOG.info("Unable to grab image and checksum information for" + tool.getNamespace() + '/' + tool.getName());
+                    }
+                    return tags;
+                }
+            } else {
+                LOG.info("Could not get response from GitLab");
```

Comment:
Might become more clear to me when I read the tests. 
But in the hypothetical where a GitLab is slow/under-load, will this return an empty list and then erase existing records?